examples/spmm/spmm.cc

```diff
@@ -179,17 +179,26 @@ std::ostream &operator<<(std::ostream &os, const Key<Rank> &key) {
   return os;
 }
 
+inline int tile2rank(int i, int j, int P, int Q)
+{
+  int p = (i % P);
+  int q = (j % Q);
+  int r = (q * P) + p;
+  return r;
+}
+
 // flow data from an existing SpMatrix on rank 0
 template <typename Blk = blk_t>
 class Read_SpMatrix : public Op<Key<2>, std::tuple<Out<Key<2>, Blk>>, Read_SpMatrix<Blk>, void> {
  public:
   using baseT = Op<Key<2>, std::tuple<Out<Key<2>, Blk>>, Read_SpMatrix<Blk>, void>;
 
-  Read_SpMatrix(const char *label, const SpMatrix<Blk> &matrix, Edge<Key<2>> &ctl, Edge<Key<2>, Blk> &out, const int P,
-                const int Q)
+  Read_SpMatrix(const char *label, const SpMatrix<Blk> &matrix, Edge<Key<2>> &ctl, Edge<Key<2>, Blk> &out, int P, int Q)
       : baseT(edges(ctl), edges(out), std::string("read_spmatrix(") + label + ")", {"ctl"}, {std::string(label) + "ij"},
               [Q](const Key<2> &key) {
-                int r = (int)key[0] * Q + (int)key[1];
+                int i = key[0];
+                int j = key[1];
+                int r = i * Q + i;
```

Comment:
Shouldn't this be tile2rank(i, j)? Or shouldn't just read_spmatrix take a keymap like the others?

Or at least, shouldn't j be part of the formula here?